defaultclient.go

```diff
@@ -761,26 +731,56 @@ func (client *DefaultClient) GetRolePermissions(roleID string, opts ...Option) (
 	return perms, err
 }
 
-// getClientInformation get client base URI
-// need client access token for authorization
-// nolint: funlen
-func (client *DefaultClient) getClientInformation(getClientInformationURL string, opts ...Option) (err error) {
+// GetClientInformation gets IAM client information,
+// it will look into cache first, if not found then fetch it to IAM.
+// forceFetch: ignore the cache and always get the latest from IAM.
+func (client *DefaultClient) GetClientInformation(namespace string, clientID string, forceFetch bool, opts ...Option) (clientInfo *ClientInformation, err error) {
 	options := processOptions(opts)
-	span, _ := jaeger.StartSpanFromContext(options.jaegerCtx, "client.ValidateAccessToken")
+	span, _ := jaeger.StartSpanFromContext(options.jaegerCtx, "client.GetClientInformation")
 
 	defer jaeger.Finish(span)
 
-	clientInformation := struct {
-		BaseURI string `json:"baseUri"`
-	}{}
+	if forceFetch {
```

Comment:
is there any case that make us to force calling the get client endpoint?

Reply:
let's make it simple first

Reply:
sure, thanks!